### DISCUSSION TITLE
Remix won't build with Puppeteer

### DISCUSSION THREAD

**Discussion by tavoyne** [remix-run/remix#3506] [Q&A] [answered]
I'm trying to use Puppeteer to generate PDFs in a `loader` function, but Remix won't compile.

```ts
import puppeteer from "puppeteer";

export const loader: LoaderFunction = async () => {
  console.log(puppeteer);
};
```

Running `remix dev` or `remix build` produces the following errors:

<img width="1064" alt="image" src="https://user-images.githubusercontent.com/29731597/174440110-cedf1fcb-3f1f-4e25-997b-f22959d312b4.png">

Node version: `16.15.1` (LTS)
Puppeteer version: `14.4.1`
Remix version: `1.5.1`

Note that everything works fine when the Puppeteer code is moved to a standalone file and executed through `node`.

Any idea what's going wrong here?

**Comment by sergiodxa** [comment, marked as answer]
Try to move the puppeteer import to a .server file and import in your routes from there.

**Reply by tavoyne** [reply]
YES, that worked.
